matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/verification/VerificationTransportRoomMessage.kt

```diff
@@ -244,7 +245,26 @@ internal class VerificationTransportRoomMessage(
                 sessionId = sessionId,
                 event = event
         ))
-        enqueueSendWork(workerParams)
+        val enqueueInfo = enqueueSendWork(workerParams)
+
+        val workLiveData = workManagerProvider.workManager
+                .getWorkInfosForUniqueWorkLiveData(uniqueQueueName())
+        val observer = object : Observer<List<WorkInfo>> {
+            override fun onChanged(workInfoList: List<WorkInfo>?) {
+                workInfoList
+                        ?.filter { it.state == WorkInfo.State.SUCCEEDED }
+                        ?.firstOrNull { it.id == enqueueInfo.second }
+                        ?.let { _ ->
+                            onDone?.invoke()
+                            workLiveData.removeObserver(this)
+                        }
+            }
+        }
+
+        // TODO listen to DB to get synced info
+        GlobalScope.launch(Dispatchers.Main) {
```

Comment:
Why using coroutines here? if it's just for dispatching to ui, prefer posting to handler directly.